emcc.py

```diff
@@ -1246,7 +1248,7 @@ def check(input_file):
 
       if shared.Settings.WASM_BACKEND:
         options.js_opts = None
-        shared.Settings.WASM = 1
```

Comment:
Add an explanation here I think, "wasm backend only supports wasm" or something

Reply:
Remove this assert since its redundant now.